src/main/java/dev/j3fftw/litexpansion/Events.java

```diff
@@ -161,101 +158,95 @@ public void onDye(PlayerInteractEntityEvent e) {
     }
 
     @EventHandler
-    public void onMiningDrillUse(PlayerInteractEvent e) {
-        final Block block = e.getClickedBlock();
+    public void onDrillUse(PlayerInteractEvent event) {
+        final Block block = event.getClickedBlock();
         if (block == null) {
             return;
         }
 
+
         final Material blockType = block.getType();
         final Location blockLocation = block.getLocation();
+        final ItemStack hand = event.getItem();
+
+        if (!SlimefunTag.STONE_VARIANTS.isTagged(blockType)) {
+            return;
+        }
 
         final MiningDrill miningDrill = (MiningDrill) SlimefunItem.getById(Items.MINING_DRILL.getItemId());
+        if (miningDrill.isItem(hand)) {
+            Validate.notNull(miningDrill);
+            if (!Check(miningDrill, event, blockLocation)) {
+                return;
+            }
 
-        Validate.notNull(miningDrill, "Can not be null");
-        if (miningDrill.isItem(e.getItem())
-            && SlimefunTag.STONE_VARIANTS.isTagged(blockType)
-            && !miningDrill.isDisabled()
-            && Slimefun.getProtectionManager().hasPermission(e.getPlayer(),
-            blockLocation, Interaction.BREAK_BLOCK)
-        ) {
-            e.setCancelled(true);
+            DrillUse(0.5f, block, blockType, blockLocation, event);
+        }
 
-            final SlimefunItem slimefunItem = BlockStorage.check(block);
+        final MiningDrill diamondDrill = (MiningDrill) SlimefunItem.getById(Items.DIAMOND_DRILL.getItemId());
+        if (diamondDrill.isItem(hand)) {
 
-            if (slimefunItem == null && ((Rechargeable) SlimefunItem.getByItem(e.getItem()))
-                .removeItemCharge(e.getItem(), 0.5F)
-            ) {
-                // This allows other plugins to register broken block as player broken
-                BlockBreakEvent newEvent = new BlockBreakEvent(block, e.getPlayer());
-                Bukkit.getServer().getPluginManager().callEvent(newEvent);
-                block.setType(Material.AIR);
-                e.getPlayer().playSound(blockLocation, Sound.BLOCK_STONE_BREAK, 1.5F, 1F);
-
-                // This has to be done because the item in the main hand is not a pickaxe
-                if (blockType == Material.STONE) {
-                    blockLocation.getWorld().dropItem(blockLocation,
-                        new ItemStack(Material.COBBLESTONE)
-                    );
-
-                } else {
-                    blockLocation.getWorld().dropItem(blockLocation,
-                        new ItemStack(blockType)
-                    );
-                }
+            Validate.notNull(diamondDrill);
+            if (!Check(diamondDrill, event, blockLocation)) {
+                return;
+            }
+
+            if (!SlimefunTag.MINEABLE_PICKAXE.isTagged(blockType)) {
```

Comment:
Not sure here, but this looks redundant as you cant get here without being included in STONE_VARIANTS.

Reply:
MINEABLE_PICKAXE includes all blocks that the pickaxe can break having it redundant wont actually matter to much but i will move it to inside the miningDrill